e2e/smoke_test.go

```diff
@@ -0,0 +1,97 @@
+package e2e
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/pkg/api"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/apis/extensions/v1beta1"
+)
+
+func TestSmoke(t *testing.T) {
+	// 1. create nginx deployment
+	di, _, err := api.Codecs.UniversalDecoder().Decode(nginxDep, nil, &v1beta1.Deployment{})
+	if err != nil {
+		t.Fatalf("unable to decode deployment manifest: %v", err)
+	}
+	d, ok := di.(*v1beta1.Deployment)
+	if !ok {
+		t.Fatalf("expected manifest to decode into *api.Service, got %T", di)
+	}
+	_, err = client.ExtensionsV1beta1().Deployments(namespace).Create(d)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.ExtensionsV1beta1().Deployments(namespace).Delete("nginx-deployment", &metav1.DeleteOptions{})
+
+	// 2. Get the nginx pod IP.
+	var p *v1.Pod
+	getPod := func() error {
+		p, err = client.CoreV1().Pods(namespace).Get("nginx", metav1.GetOptions{})
+		if err == nil && p.Status.Phase == v1.PodRunning {
+			t.Logf("waiting on nginx deployment: %v", err)
+			return fmt.Errorf("pod not yet running: %v", err)
+		}
+		return nil
+	}
+	if err := retry(10, time.Second*10, getPod); err != nil {
+		t.Fatalf("timed out waiting for nginx pod: %v", err)
+	}
+
+	// 3. Create a wget pod that hits the nginx pod IP.
+	wgetPod.Spec.Containers[0].Command = []string{"wget", p.Status.PodIP}
+
+	_, err = client.CoreV1().Pods(namespace).Create(wgetPod)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.CoreV1().Pods(namespace).Delete(wgetPod.ObjectMeta.Name, &metav1.DeleteOptions{})
+
+	getPod = func() error {
+		p, err = client.CoreV1().Pods(namespace).Get("wget-pod", metav1.GetOptions{})
+		if err == nil && p.Status.Phase == v1.PodSucceeded {
+			t.Logf("waiting on wget-pod: %v", err)
+			return fmt.Errorf("pod not yet running: %v", err)
```

Comment:
Oh, good catch. I completely got this logic reversed when I copy-pasted your code. I think I my mind saw an if statement and was tunnel vision about it being an error path. Fixing.